src/core/namespace.c

```diff
@@ -1972,6 +2070,16 @@ static int apply_mounts(
                 return log_debug_errno(r, "Failed to open /proc/self/mountinfo: %m");
         }
 
+        /* Prepare a list of verity metadata, that will be used to check the root hash of each image and
+         * drop duplicates. Overlayfs rejects supplying the same directory inode twice as determined by
+         * filesystem UUID and file handle in lowerdir=, even if they are mounted on different paths, as it
+         * resolves each mount to its source filesystem. */
+        extension_verities = new0(VeritySettings, p->n_extension_images);
```

Comment:
please allocate this only if p->n_extension_images is > 0, after all this is going to be the exception, not the rule in many cases.